Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -398,9 +406,44 @@ open class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synchro
         // clients since the beginning of time instead of looking at `self.lastFetched`.
         return clientsClient.getSince(0)
             >>== { response in
-                return self.wipeIfNecessary(localClients)
-                    >>> { self.applyStorageResponse(response, toLocalClients: localClients, withServer: clientsClient, notifier: notifier) }
+                return self.maybeDeleteClients(response: response, withServer: storageClient)
+                    >>> { self.wipeIfNecessary(localClients)
+                                >>> { self.applyStorageResponse(response, toLocalClients: localClients, withServer: clientsClient, notifier: notifier) }
+                    }
+
             }
             >>> { deferMaybe(self.completedWithStats) }
     }
+
+    func maybeDeleteClients(response: StorageResponse<[Record<ClientPayload>]>, withServer storageClient: Sync15StorageClient) -> Success {
```

Comment:
Q: Is this a method that will be used outside or can we make it private to avoid confusion with its name of maybe deleting client?